.github/workflows/upload-win-installer.yml

```diff
@@ -0,0 +1,96 @@
+name: Upload Windows Installer
+
+on:
+  release:
+    types: [created, published, edited]
+  workflow_dispatch:
+    inputs:
+      version:
+        description: 'Release version to build and upload (e.g. "4.2.1")'
+        required: true
+jobs:
+  build:
+    runs-on: windows-latest
+    env:
+        FETCH_BASE_URL: ${{ github.server_url }}/${{ github.repository }}
+    steps:
+    - name: Determine version
+      id: getversion
+      run: |
+        $version = "${{ inputs.version }}"
+        if ($version.Length -lt 1) {
+          $version = "${{ github.event.release.tag_name }}"
+          if ($version.Length -lt 1) {
+            Write-Host "Could not determine version!"
+            Exit 1
+          }
+        }
+        Write-Output "::set-output name=version::$version"
+    - uses: actions/checkout@v3
+    - name: Check
+      id: check
+      run: |
+        Push-Location contrib\win-installer
+        .\check.ps1 ${{steps.getversion.outputs.version}}
+        $code = $LASTEXITCODE
+        if ($code -eq 2) {
+          Write-Output "::set-output name=already-exists::true"
+          Pop-Location
+          Exit 0
+        }
+        Write-Output "UPLOAD_ASSET_NAME=$env:UPLOAD_ASSET_NAME" | Out-File -FilePath $env:GITHUB_ENV -Append
+        Pop-Location
+        Exit $code
+    - name: Set up Go
+      uses: actions/setup-go@v3
+      if: steps.check.outputs.already-exists != 'true'
+      with:
+        go-version: 1.18
```

Comment:
This changes maybe on a yearly basis, with the 'source of truth' being in our CI VM and Container-image build process.  It's going to be really difficult for developers to remember to update the value in this file.  Any ideas on if/how this hard-coded version could be looked up or retrieved somehow?

Reply:
The value is also differs over-time WRT release-branches.  i.e. we don't want the workflow hard-coding `golang 1.39` while we try and do a release for an old `podman v4.4` that relied on `golang 1.18` (for example).

One idea could be: Setup the Cirrus-CI artifacts task to publish the version of golang that was used into a text file.  That could then be curled from a release-branch specific URL.  Downside is it's more complex machinery ontop of complex machinery :(

Reply:
Another idea: Make the golang version a `workflow_dispatch` input, so it can be changed manually when building for an old release branch.

Reply:
It's a good point. This go version is only used to compile two go artifacts for the project (a dll and an exe). However those artifacts are reusing the same pkg code and the module definition from the wider podman project. So while it doesn't need to precisely match, it should be >= the go version in the module.  I like your idea of providing a defaulted override to the workflow_dispatch. I can add that.